corehq/ex-submodules/casexml/apps/case/mock/case_block.py

```diff
@@ -25,7 +25,11 @@ class CaseBlock(object):
     def __init__(self, case_id, date_modified=None, user_id=undefined,
                  owner_id=undefined, external_id=undefined, case_type=undefined,
                  case_name=undefined, create=False, date_opened=undefined, update=None,
-                 close=False, index=None, strict=True):
+                 close=False, index=None, strict=True, date_opened_deprecated_behavior=False):
+        """
+        When `date_opened_deprecated_behavior`, a date_opened YYYY-MM-DD value is inserted on new cases.
+        This is deprecated behavior, because it prevents the superior default behavior from kicking in.
```

Comment:
where does this other default behaviour happen?

Reply:
We do it explicitly during case processing. In the sql backend for example I think it'd done here: https://github.com/dimagi/commcare-hq/blob/f7b136fb44787e5fdacde7dce5c922880e9a01c8/corehq/form_processor/backends/sql/update_strategy.py#L138-L139